spec/spec_helper.rb

```diff
@@ -25,14 +25,29 @@
 
 require 'vcr'
 
+vcr_record_mode = ENV["VCR"] ? :all : :none
+
 VCR.configure do |config|
   config.cassette_library_dir = "spec/fixtures/vcr_cassettes"
   config.hook_into :webmock
   config.configure_rspec_metadata!
-  config.default_cassette_options = { match_requests_on: [:query] }
-
-  record_mode = ENV["VCR"] ? ENV["VCR"].to_sym : :new_episodes
-  config.default_cassette_options = { record: record_mode }
+  config.default_cassette_options = { record: vcr_record_mode }
+  config.default_cassette_options = { match_requests_on: [
+    :method,
+    :query,
+    :path,
+    :body,
+    :paging_headers
+  ] }
+
+  config.register_request_matcher :paging_headers do |r1, r2|
```

Comment:
Not sure about this style - shouldn't the paging_headers constant be PAGING_HEADERS? 
and unsure whether deleting items from the headers hash with keep_if is a good idea - isn't this just the same as r1.headers.select { |k, _| paging_headers.include?(k) } == r2.headers.select { |k, _| paging_headers.include?(k) } ?